### DISCUSSION TITLE
PropertyGrid控件 double属性异常

### DISCUSSION THREAD

**Discussion by yezi15623** [HandyOrg/HandyControl#1397] [Q&A]
PropertyGrid控件绑定的对象中有double类型属性时, 控件无法输入小数点。以及float类型属性显示异常 如0.5 显示为0.5000000154这样
![image](https://github.com/HandyOrg/HandyControl/assets/75667873/c31cf2cd-ea0f-4aea-8788-c204379b3223)
![image](https://github.com/HandyOrg/HandyControl/assets/75667873/48dbce5e-7626-49bc-845c-94b5a6e817cf)

**Comment by nhk035** [comment]
最近刚学习wpf，也碰到了这个问题。看了一下代码，应该是这么回事。
PropertyGrid里面的数值编辑器使用了windows的数值显示框[NumericUpDown](https://learn.microsoft.com/zh-cn/dotnet/api/system.windows.forms.numericupdown?view=windowsdesktop-6.0)，这个数值显示框的格式由[DecimalPlaces](https://learn.microsoft.com/zh-cn/dotnet/api/system.windows.forms.numericupdown.decimalplaces?view=windowsdesktop-6.0)来控制，默认只显示小数点之前。
所以如果想要输入浮点数可以把原来的控件代码修改一下，然后引用重新编译之后的dll。
路径：HandyControl\src\Shared\HandyControl_Shared\Controls\PropertyGrid\Editors\NumberPropertyEditor.cs

``` csharp
public override FrameworkElement CreateElement(PropertyItem propertyItem) => new NumericUpDown
    {
        IsReadOnly = propertyItem.IsReadOnly,
        DecimalPlaces = 2,
        Increment = 0.01,
        Minimum = Minimum,
        Maximum = Maximum
    };
```
这里我是设置成了保留两位小数，每个增量为0.01。不过这样改了以后所有的数值编辑框全部都变成了两位小数，反正是能用了。各位大佬如果有更好的办法，帮忙解答一下。
